introduction-to-datasets.md

```diff
@@ -0,0 +1,130 @@
+
+[list]: {{site.baseurl}}/apidocs/org/kitesdk/data/Datasets.html#list(java.net.URI)
+[uris]:{{site.baseurl}}/URIs.html
+
+## Schemas
+
+A schema defines the field names and datatypes for a dataset. Kite relies on an Apache Avro schema definition for each dataset. Kite standardizes data definition by using Avro schemas for both Parquet and Avro, and supports the standard Avro object models _generic_ and _specific_.
+
+For example, this is the schema definition for a table using four columns from the `movies` dataset.[<sup>1</sup>](#notes)
+
+```json
+{
+  "type":"record",
+  "name":"Movie",
+  "namespace":"org.kitesdk.examples.data",
+  "fields":[
+    {"name":"id","type":"int"},
+    {"name":"title","type":"string"},
+    {"name":"release_date","type":"string"},
+    {"name":"imdb_url","type":"string"}
+  ]
+}
+```
+
+The following links provide examples for inferring schemas from data files or Java classes.
+
+| Java API                                                      | Command Line Interface |
+| --------                                                      | ---------------------- |
+| [Inferring a schema from a Java Class][api-schema-from-class] | [Inferring a schema from a Java class][cli-schema-from-class] |
+| [Using the schema of an Avro data file][api-schema-from-data] | [Inferring a schema from a CSV file][cli-schema-from-csv] |
+
+[api-schema-from-class]: {{site.baseurl}}/Inferring-a-Schema-from-a-Java-Class.html
+[api-schema-from-data]: {{site.baseurl}}/Inferring-a-Schema-from-an-Avro-Data-File.html
+[cli-schema-from-class]: {{site.baseurl}}/cli-reference.html#obj-schema
+[cli-schema-from-csv]: {{site.baseurl}}/cli-reference.html#csv-schema
+
+## Partitions
+
+Partitions define logical divisions for data storage. For example, you might most often work with data using time-based queries. You can define a partitioning strategy by year, month, and day. When you are using data from January 8, 2015, Hadoop only has to access data stored in the partition `/year=2015/month=1/day=8`. By using partitions that correspond to your most common queries, your applications run more quickly.
+
+Partitioning is optional, because there are times when partitioning is not the most efficient solution. However, you should always consider partitioning as a best practice when planning your dataset.
+
+You define your partition strategy in [partition strategy JSON format][psjf] and apply it when you create your dataset. See [Partitioned Datasets][ps].
+
+When using the API, you define your dataset in the [`DatasetDescriptor.Builder`][ddb]. 
+
+For example, you can use the CLI command [`partition-config`][clipc] to define a partition strategy using name:value pairs to specify that the dataset should be partitioned on a timestamp (`ts`) field by year, month, and day.
+
+```
+user@work:~$ kite-dataset partition-config ts:year ts:month ts:day -s rating.avsc
```

Comment:
Added:

"Kite validates your partition strategy using the dataset schema. This catches problems early. For example, if you mistakenly type in your partition definition _timestmap_, validation catches the typo because the schema has a _timestamp_ column. Kite also validates that field types can be partitioned on the selected value. For example, you can't extract a year from a string."